miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/RunScannerClient.java

```diff
@@ -0,0 +1,113 @@
+      .name("miso_runscanner_client_save_new").help("The number of times a new run was found.").register();
+
+
+  private static final LatencyHistogram saveTime = new LatencyHistogram("miso_runscanner_client_save_time",
+      "Time to save a run (in seconds).");
+
+  private static final Counter saveUpdate = Counter.build()
+      .name("miso_runscanner_client_save_update").help("The number of times a run was found in the database.").register();
+
+  private static final Gauge scanTimestamp = Gauge.build()
+      .name("miso_runscanner_client_scan_timestamp").help("The UNIX time when the client last attempted to scan the servers.").register();
+
+  private static final Counter serverFailures = Counter.build()
+      .name("miso_runscanner_client_read_errors").help("The number of times the server failed to respond.").labelNames("url").register();
+
+  private static final LatencyHistogram serverReadTime = new LatencyHistogram("miso_runscanner_client_read_time",
+      "Time to download updates from a server in seconds.", "url");
+
+  private static final Pattern WHITESPACE = Pattern.compile("\\s+");
+
+  // This self-autowires so we can call the proxified version of ourselves to create transactions.
+  @Autowired
+  private RunScannerClient client;
+  private final Semaphore lock = new Semaphore(1);
+  @Autowired
+  private RunService runService;
+  private final ConcurrentMap<String, ProgressiveRequestDto> servers = new ConcurrentHashMap<>();
+
+  public void process(List<NotificationDto> results) {
+    try (AutoCloseable timer = acquireTime.start()) {
+      lock.acquire();
+    } catch (Exception e) {
+      log.error("Failed to acquire lock", e);
+      return;
+    }
+    for (NotificationDto dto : results) {
+      try (AutoCloseable timer = saveTime.start()) {
+        (runService.processNotification(dto) ? saveNew : saveUpdate).inc();
+        saveCount.inc();
+      } catch (Exception e) {
+        log.error("Failed to save run: " + dto.getRunAlias(), e);
+        saveFailures.inc();
+      }
+    }
+    lock.release();
+  }
+
+  @Scheduled(fixedDelay = 5 * 60 * 1000)
+  public void scheduler() {
+    scanTimestamp.set(System.currentTimeMillis() / 1000);
+    RestTemplate template = new RestTemplate();
+    List<NotificationDto> results = new ArrayList<>();
+    for (Entry<String, ProgressiveRequestDto> entry : servers.entrySet()) {
+      try (AutoCloseable timer = serverReadTime.start(entry.getKey())) {
+        ProgressiveResponseDto response = template.postForObject(entry.getKey() + "/runs/progressive", entry.getValue(),
+            ProgressiveResponseDto.class);
+        entry.getValue().update(response);
+        results.addAll(response.getUpdates());
+      } catch (Exception e) {
+        log.error("Failed to get runs from " + entry.getKey(), e);
+        serverFailures.labels(entry.getKey()).inc();
+      }
+    }
+    // Perform this in a transaction
+    client.process(results);
```

Comment:
Why control the transaction here? RunService will already create a separate transaction for each run save. If one run fails to save, I don't think it should affect other runs?